analyses/tumor-purity-exploration/01_explore-tumor-purity.Rmd

```diff
@@ -0,0 +1,503 @@
+---
+title: "Tumor purity exploration"
+author: "SJ Spielman for CCDL"
+date: "2023"
+output: 
+  html_notebook:
+    toc: TRUE
+    toc_float: TRUE
+params:
+  release: release-v23-20230115
+---
+
+```{r setup, include=FALSE}
+library(tidyverse)
+theme_set(theme_bw())
+```
+This notebook performs the following:
+
+- Explores the distribution of tumor purity values as well as their relationship to other metadata values
+- Identifies RNA-level samples for which we reliably know their tumor purity based on associated WGS data
+- Explores potential tumor purity thresholds that could be applied in other contexts to assess the influence of tumor purity on certain RNA-level analyses
+
+## Set directories and files
+
+```{r}
+# Input
+root_dir <- rprojroot::find_root(rprojroot::has_dir(".git"))
+data_dir <- file.path(root_dir, "data")
+pal_dir <- file.path(root_dir, "figures", "palettes")
+metadata_file <- file.path(data_dir, "pbta-histologies.tsv")
+pal_file <- file.path(pal_dir, "broad_histology_cancer_group_palette.tsv")
+
+# Output
+results_dir <- file.path("results")
+if (!(dir.exists(results_dir))) {
+  dir.create(results_dir)
+}
+rna_stranded_df_file <- file.path(results_dir, "metadata_rna_stranded_same-extraction.tsv")
+rna_stranded_thresholded_df_file <- file.path(results_dir, "metadata_thresholded_rna_stranded_same-extraction.tsv")
+```
+
+
+Read in:
+
+```{r}
+metadata <- read_tsv(metadata_file) 
+palette_mapping_df <- read_tsv(pal_file)
+```
+
+## Exploration of `tumor_fraction`
+
+The quantity `tumor_fraction` provides information about what percent of the sample was likely tumor. 
+We'll explore this variable:
+
+```{r}
+# Note that this quantity is only recorded, if ever, with WGS libraries: 
+presence_table <- table(metadata$experimental_strategy,
+                        is.na(metadata$tumor_fraction))
+colnames(presence_table) <- c("Tumor fraction known", "Tumor fraction unknown")
+presence_table
+```
```

Comment:
Not a strongly held opinion, but this doesn't seem that useful to me.

Reply:
This table was helpful for me while coding this up see literally do we know _anything_ about tumor purity in the first place, so I prefer to keep it.